src/h2load.cc

```diff
@@ -407,9 +486,17 @@ int Client::make_socket(addrinfo *addr) {
 int Client::connect() {
   int rv;
 
-  record_client_start_time();
-  clear_connect_times();
-  record_connect_start_time();
+  if (worker->config->warm_up_time == 0 
+    || worker->current_phase == MAIN_DURATION) {
```

Comment:
Is this really conditioning on whether we are running with a warmup_time set or not? I feel the code is hard to read here as it talks about a bunch of phases, and it's unclear to me under which conditions we actually might be in them. Do clients get connected in main duration after a warmup expires? Etc.